apps/emqx_rule_engine/src/emqx_rule_metrics.erl

```diff
@@ -289,61 +314,74 @@ handle_info(ticking, State = #state{rule_speeds = undefined}) ->
 
 handle_info(_Info, State) ->
     {noreply, State}.
 
-code_change({down, Vsn}, State = #state{metric_ids = MIDs}, _Extra)
-        when Vsn =:= "4.2.0";
-             Vsn =:= "4.2.1" ->
-    emqx_metrics:ensure('actions.failure'),
-    emqx_metrics:set('actions.failure',
-                     emqx_metrics:val('actions.error')
-                     + emqx_metrics:val('actions.exception')),
-    [begin
-        Matched = get_rules_matched(Id),
-        Succ = get_actions_success(Id),
-        Error = get_actions_error(Id),
-        Except = get_actions_exception(Id),
-        ok = delete_counters(Id),
-        ok = create_counters(Id),
-        inc_rules_matched(Id, Matched),
-        inc_actions_success(Id, Succ),
-        inc_actions_error(Id, Error + Except)
-    end || Id <- sets:to_list(MIDs)],
-    {ok, State};
-
-code_change(Vsn, State = #state{metric_ids = MIDs}, _Extra)
-        when Vsn =:= "4.2.0";
-             Vsn =:= "4.2.1" ->
-    [emqx_metrics:ensure(Name)
-     || Name <-
-        ['actions.error', 'actions.taken',
-         'actions.exception', 'actions.retry'
-        ]],
-    emqx_metrics:set('actions.error', emqx_metrics:val('actions.failure')),
-    [begin
-        Matched = get_rules_matched(Id),
-        Succ = get_actions_success(Id),
-        Error = get_actions_error(Id),
-        ok = delete_counters(Id),
-        ok = create_counters(Id),
-        inc_rules_matched(Id, Matched),
-        inc_actions_success(Id, Succ),
-        inc_actions_error(Id, Error)
-    end || Id <- sets:to_list(MIDs)],
-    {ok, State};
+code_change({down, _Vsn}, State = #state{metric_ids = MIDs}, [Vsn]) ->
+    case string:tokens(Vsn, ".") of 
+        ["4", "3", SVal] ->
+            {Val, []} = string:to_integer(SVal),
+            case Val  =< 7 of
+                true ->
+                    [begin
+                        Passed = get_rules_passed(Id),
+                        Take = get_actions_taken(Id),
+                        Success = get_actions_success(Id),
+                        Error = get_actions_error(Id),
+                        Exception = get_actions_exception(Id),
+                        Retry = get_actions_retry(Id),
+                        ok = delete_counters(Id),
+                        ok = create_counters(Id, 7),
+                        inc_rules_matched(Id, Passed),
+                        inc_actions_taken(Id, Take),
+                        inc_actions_success(Id, Success),
+                        inc_actions_error(Id, Error),
+                        inc_actions_exception(Id, Exception),
+                        inc_actions_retry(Id, Retry)
+                    end || Id <- sets:to_list(MIDs)],
+                    {ok, State};
+                false -> {ok, State}
+            end;
+        _ -> {ok, State}
+    end;
+
+code_change(_Vsn, State = #state{metric_ids = MIDs}, [Vsn]) ->
+    case string:tokens(Vsn, ".") of
+        ["4", "3", SVal] ->
```

Comment:
We should change it to "4.4.0". The only version to be upgraded from is "4.4.0" (current version is 4.4.1)

Reply:
we'll create another PR to fix this.